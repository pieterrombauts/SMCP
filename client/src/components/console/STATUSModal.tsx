import React from 'react';
import styled from 'styled-components';
import 'bootstrap/dist/css/bootstrap.min.css';
import { STATUS } from './STATUS'

interface AppProps {
  className?: string;
  show: boolean;
  closeFunction: () => void;
};

const USTATUSModal: React.FC<AppProps> = ( props ) => {
  function handleOverlayClick(event: React.MouseEvent<HTMLDivElement>) {
    if (event.target === event.currentTarget) {
      props.closeFunction();
    }
  }
  return(
    <div id="status-modal" className={props.className}>
      <div id="modal-overlay-status" onMouseUp={handleOverlayClick}>
        <div id="modal-content-status">

          


          <STATUS />
        </div>
      </div>
    </div>
  );
}

export const STATUSModal = styled(USTATUSModal)`
  visibility: hidden;
  #modal-overlay-status {
    position: absolute;
    top: 0;
    left: 0;
    width: 100%;
    height: 100%;
    z-index: 9;
    background-color: rgba(0,0,0,0.5);
  }

  #modal-content-status {
    background-color: #FFF;
    position: absolute;
    width: 1500px;
    height: 70%;
    left: 50%;
    top: 50%;
    transform: translate(-50%, -50%);
    border-radius: 10px;
    padding: 20px;
    z-index: 10;
  }
`;