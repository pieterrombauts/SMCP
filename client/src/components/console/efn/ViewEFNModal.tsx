import React from 'react';
import styled from 'styled-components';
import 'bootstrap/dist/css/bootstrap.min.css';
import { EFNDetails } from './EFNDetails'
import { statusReport } from '../customTypes';

interface AppProps {
  className?: string;
  show: boolean;
  closeFunction: () => void;
  reports: statusReport[];
  selectedEfn: string;
  lobbyID: string;
};

export const UViewEFNModal: React.FC<AppProps> = ( props ) => {
  function handleOverlayClick(event: React.MouseEvent<HTMLDivElement>) {
    if (event.target === event.currentTarget) {
      props.closeFunction();
    }
  }
  return(
    <div id="view-efn-modal" className={props.className}>
      <div id="modal-overlay-view-efn" onMouseUp={handleOverlayClick}>
        <div id="modal-content-view-efn">
          <EFNDetails lobbyID={props.lobbyID} reports={props.reports} efnID={props.selectedEfn} closeFunction={props.closeFunction}/>
        </div>
      </div>
    </div>
  );
}

export const ViewEFNModal = styled(UViewEFNModal)`
  visibility: hidden;
  #modal-overlay-view-efn {
    position: absolute;
    top: 0;
    left: 0;
    width: 100%;
    height: 100%;
    z-index: 9;
    background-color: rgba(0,0,0,0.5);
  }

  #modal-content-view-efn {
    background-color: #FFF;
    position: absolute;
    width: 40%;
    left: 50%;
    top: 50%;
    transform: translate(-50%, -50%);
    border-radius: 10px;
    padding: 40px;
    z-index: 10;
  }
`;