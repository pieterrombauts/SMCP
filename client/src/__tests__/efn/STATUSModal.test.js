import React from "react";
import 'jest-styled-components'
import toJson from 'enzyme-to-json'
import { USTATUSModal } from '../../components/console/efn/STATUSModal'
import { shallow, mount, render} from 'enzyme';

describe("EFN modal popup", () => {
  it('renders correctly', () => {
    const wrapper = shallow(<USTATUSModal show={true} closeFunction={jest.fn()} lobbyID={'111111'} userRole={'spartan'}/>)
    const tree = toJson(wrapper);
    expect(tree).toMatchSnapshot();
  })
})