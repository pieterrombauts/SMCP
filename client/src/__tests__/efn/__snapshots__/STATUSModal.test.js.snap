// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EFN modal popup renders correctly 1`] = `
<div
  id="status-modal"
>
  <div
    id="modal-overlay-status"
    onMouseUp={[Function]}
  >
    <div
      id="modal-content-status"
    >
      <Connect(Styled(USTATUS))
        closeFunction={[MockFunction]}
        lobbyID="111111"
        userRole="spartan"
      />
    </div>
  </div>
</div>
`;
